db/migrate/20170907215045_add_details_to_endpoints.rb

```diff
@@ -0,0 +1,8 @@
+class AddDetailsToEndpoints < ActiveRecord::Migration[5.1]
+  def change
+    add_column :endpoints, :endpoint_node, :string, null: false
+    add_column :endpoints, :endpoint_storage, :string, null: false
```

Comment:
This seems like a bad name - what IS endpoint_storage?

Reply:
good question, i had to look that up again just now, which would sort of indicate an unclear name.  but it is this nebulous thing that could be a file path, or an S3 bucket, or a ??? for some different storage technology.  see @julianmorley's diagram here:  https://user-images.githubusercontent.com/8161286/29797985-0ab543f0-8c0f-11e7-836f-487c3d347a40.png

i'm not sure what would be a better name, though i'm open to suggestions.  one thought i had was that we could collapse `endpoint_node` and `endpoint_storage` into a single `endpoint_uri` column.  and so, julian's example might turn into something like `sul-sdr-services:/services-disk03/sdr2objects/` and `s3.us-east-2.amazonaws.com/sdr-bucket-01` if you were to convert the first couple rows of his example table to that approach.  the two things that make me hesitant to do that are:  1) is there a standard way to construct an S3 URI like that?  i imagine there is, but i don't know if offhand, since i'm unfamiliar w/ the workings of S3 and AWS.  2) it'd be harder to query that setup in a performant way (e.g., give me all the MOABs that are archived to sul-dlss.oraclecloud.com, and give me all the MOABs that have online copies on sul-sdr-services, would both require the query to do some text wrangling if the two cols were combined into one, and text wrangling is slow, though maybe that's the sort of info that gets shoved into argo's solr index anyway, and then you can do faceting or substring search or whatever in a more performant way without going directly to postgres).

Reply:
after discussion w/ @tingulfsen, we decided that `endpoint_storage_location` was better.  i might even be inclined to shorten that to `storage_location`?

Reply:
I like `storage_location` and can live with `endpoint_storage_location`